From 7ea8741ac6428c5db3c3562343794d34e49fafe1 Mon Sep 17 00:00:00 2001
From: Erez Shaul <eshaul@habana.ai>
Date: Tue, 17 Sep 2024 17:33:02 +0300
Subject: [PATCH] [SW-201625] i2c_indirect: adjust ram_size naming

Since SRAM_SIZE in no longer exists in G3 embbedded specs, and it is
used in a common code, it need to be adjusted accordingly, so in case
it exists, it will be used, otherwize, SDRAM_SIZE will be used.

Change-Id: Ie117c4621687439f053c80778eb5cb489f2bca26
Signed-off-by: Erez Shaul <eshaul@habana.ai>
---

diff --git a/lib/preboot/i2cslave_indirect.c b/lib/preboot/i2cslave_indirect.c
index 04caacf..b3fe54a 100644
--- a/lib/preboot/i2cslave_indirect.c
+++ b/lib/preboot/i2cslave_indirect.c
@@ -206,16 +206,22 @@
 
 }
 
+#ifdef SRAM_SIZE
+#define MAX_SIZE SRAM_SIZE
+#else
+#define MAX_SIZE SDRAM_SIZE
+#endif
+
 static void i2csi_req_memcpy_ctrl(void)
 {
 	op_memcpy.address = preboot_axi2local_translate(op_memcpy.address);
 	if (op_memcpy.address < SDRAM_BASE)
 		goto memcpy_ctrl_fail;
 
-	if (op_memcpy.address > SDRAM_BASE + SRAM_SIZE)
+	if (op_memcpy.address > SDRAM_BASE + MAX_SIZE)
 		goto memcpy_ctrl_fail;
 
-	if (op_memcpy.size > SRAM_SIZE)
+	if (op_memcpy.size > MAX_SIZE)
 		goto memcpy_ctrl_fail;
 
 	op_memcpy.status = HL_I2CI_MEMCPY_STS_PASS;
